### PR TITLE
18 login page

### PR DESCRIPTION
**Pull request by fadeomar**
create the login page that contain two filed input one for username and password 
and two button one for login and the other for back to the previous page ... 

note : the login button doesn't do any thing for now !! 

you can test it by inport the Login component in the app file 
````
import Login from './components/pages/Login';
````
and inside the ````<BrowserRouter>```` component
````
        <Route exact path="/login" component={Login} />
````

and for sure you would use Route from React :)

### CODE REVIEW

File: client/src/components/pages/Login/index.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import { loginFun, inputFun } from '../../../services/api';
+
+class Login extends Component {
+  state = {};
+
+  render() {
+    return (
+      <div className="loginPage">
+        <div className="backBtn">
```
Comment (Fatmasiam) [inline client/src/components/pages/Login/index.js]:
```suggestion
        <div className="backbtn">
```
```suggestion
        <div className="back-btn">
```

### **DON'T USE CAMEL CASE ON CLASS NAMING**

File: client/src/components/pages/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import './index.css';
+import Button from '../../common/Button';
+import Input from '../../common/Input';
+import BackButton from '../../common/BackButton';
+
+import { loginFun, inputFun } from '../../../services/api';
+
+class Login extends Component {
+  state = {};
```
Comment (Fatmasiam) [inline client/src/components/pages/Login/index.js]: we don't develop static pages !!!!

File: client/src/components/pages/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  render() {
+    return (
+      <div className="loginPage">
+        <div className="backBtn">
+          <BackButton back={() => console.log('the url for previous page  ')} />
+        </div>
+        <div className="loginBox">
```
Comment (Fatmasiam) [inline client/src/components/pages/Login/index.js]:
```suggestion
        <div className="login-box">
```

File: client/src/components/pages/Login/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+              <input
+                type="text"
+                placeholder="password"
+                className="password-input"
+                name="username-password"
+                aria-label="password"
+              /> */}
+              <Input
+                type="text"
+                name="username-passwrd"
```
Comment (Fatmasiam) [inline client/src/components/pages/Login/index.js]:
```suggestion
                name="passwrd"
```

File: client/src/components/pages/Login/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        </div>
+        <div className="loginBox">
+          <form action="/" className="login-form">
+            <div className="login-wlecome">
+              <h1>Welcome Back</h1>
+            </div>
+            <div className="login-inpputs">
+              <Input
+                type="text"
+                name="username-input"
```
Comment (Fatmasiam) [inline client/src/components/pages/Login/index.js]:
```suggestion
                name="username"
```

File: client/src/services/api.js [hunk 6]
```diff
@@ -1,3 +1,6 @@
-const fetchData = () => fetch('/api/v1').then(res => res.json());
-
-export default { fetchData };
+export const fetchData = () => fetch('/api/v1').then(res => res.json());
+export const loginFun = () =>
+  console.log('onClick function at the logiBtn element');
+export const inputFun = () =>
+  console.log('onClick function at the input element');
+export default { fetchData, loginFun, inputFun };
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/services/api.js]: export **

File: client/src/services/api.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
-const fetchData = () => fetch('/api/v1').then(res => res.json());
-
-export default { fetchData };
+export const fetchData = () => fetch('/api/v1').then(res => res.json());
+export const loginFun = () =>
+  console.log('onClick function at the logiBtn element');
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/services/api.js]: Done
.

File: client/src/services/api.js [hunk 8]
```diff
@@ -1,3 +1,6 @@
...
+export const inputFun = () =>
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/services/api.js]: Done

File: client/src/services/api.js [hunk 9]
```diff
@@ -1,3 +1,6 @@
-const fetchData = () => fetch('/api/v1').then(res => res.json());
-
-export default { fetchData };
+export const fetchData = () => fetch('/api/v1').then(res => res.json());
+export const loginFun = () =>
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/services/api.js]: Done

File: client/src/services/api.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/services/api.js]: Done

File: client/src/components/pages/Login/index.js (hunk 1 above)
Comment (fadeomar) [reply client/src/components/pages/Login/index.js]: Done

File: client/src/components/pages/Login/index.js (hunk 2 above)
Comment (fadeomar) [reply client/src/components/pages/Login/index.js]: Done

File: client/src/components/pages/Login/index.js (hunk 3 above)
Comment (fadeomar) [reply client/src/components/pages/Login/index.js]: Done

**Comment by 11janci** [comment]
pls resolve conflict

File: client/src/components/pages/Login/index.js [hunk 10]
```diff
@@ -1,7 +1,106 @@
...
+    const { username, password, errMSg } = this.state;
+    const {
+      history: { goBack },
+    } = this.props;
+    return (
+      <div className="login-page">
+        <div className="login-box">
+          <form action="/" className="login-form">
+            <div className="back-btn">
+              <BackButton back={() => goBack()} />
```
Comment (11janci) [inline client/src/components/pages/Login/index.js]:
```suggestion
              <BackButton back={goBack} />
```

File: client/src/services/api.js [hunk 11]
```diff
@@ -1,3 +1,11 @@
...
-export default { fetchData };
+export default {
+  login: data => {
+    return fetch('/api/v1/login', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    }).then(res => res.json());
```
Comment (11janci) [inline client/src/services/api.js]:
```suggestion
    }).then(res => {
        if (!res.ok) {
            throw Error(res.message ? res.message : 'Login failed');
        }
        return res.json();
    });
```
the `if (!res.ok) {` checks the HTTP status code, evaluates to true if it's anything other than success (200). Every API in this module should utilize this approach (you can extract it into a function and use `.then(handleResp)`)

File: client/src/components/pages/Login/index.js [hunk 12]
```diff
@@ -1,7 +1,106 @@
...
+    if (username && password) {
+      const {
+        history: { push },
+      } = this.props;
+      api
+        .login({
+          username,
+          password,
+        })
+        .then(res => {
```
Comment (11janci) [inline client/src/components/pages/Login/index.js]: see my comment for the api service, after that change this to
```
.then(() => push('/'))
.catch(() => this.setState({ errMSg: res.message }))
```

**Pull request merged**